source/server/data_moniker_service.cpp

```diff
@@ -0,0 +1,298 @@
+    //---------------------------------------------------------------------
+    void DataMonikerService::InitiateMonikerList(const MonikerList& monikers, EndpointList* readers, EndpointList* writers)
+    {
+        for (auto readMoniker: monikers.read_monikers())
+        {
+            auto instance = readMoniker.data_instance();
+            auto source = readMoniker.data_source();
+            auto it = _endpoints.find(source);
+            MonikerEndpointPtr ptr = nullptr;
+            if (it != _endpoints.end())
+            {
+                ptr = (*it).second;
+            }
+            readers->push_back(EndpointInstance(ptr, reinterpret_cast<void*>(instance)));
+        }
+        for (auto writeMoniker: monikers.write_monikers())
+        {
+            auto instance = writeMoniker.data_instance();
+            auto source = writeMoniker.data_source();
+            auto it = _endpoints.find(source);
+            MonikerEndpointPtr ptr = nullptr;
+            if (it != _endpoints.end())
+            {
+                ptr = (*it).second;
+            }
+            writers->push_back(EndpointInstance(ptr, reinterpret_cast<void*>(instance)));
+        }
+    }
+
+    //---------------------------------------------------------------------
+    //---------------------------------------------------------------------
+    void DataMonikerService::RunSidebandReadWriteLoop(string sidebandIdentifier, ::SidebandStrategy strategy, EndpointList* readers, EndpointList* writers)
+    {
+        std::cout << "Enter RunSidebandReadWriteLoop" << std::endl;
+
+    #ifndef _WIN32
+        if (strategy == ::SidebandStrategy::RDMA_LOW_LATENCY ||
+            strategy == ::SidebandStrategy::SOCKETS_LOW_LATENCY)
+        {
+            pid_t threadId = syscall(SYS_gettid);
+            ::SysFsWrite("/dev/cgroup/cpuset/LabVIEW_tl_set/tasks", std::to_string(threadId));
+
+            cpu_set_t cpuSet;
+            CPU_ZERO(&cpuSet);
+            CPU_SET(8, &cpuSet);
+            //CPU_SET(8, &cpuSet);
+            sched_setaffinity(threadId, sizeof(cpu_set_t), &cpuSet);
+        }
+    #endif
+
+        google::protobuf::Arena arena;
+        int64_t sidebandToken;
+        GetOwnerSidebandDataToken(sidebandIdentifier.c_str(), &sidebandToken);
+        while (true)
+        {
+            auto writeRequest = google::protobuf::Arena::CreateMessage<SidebandWriteRequest>(&arena);
+            if (!ReadSidebandMessage(sidebandToken, writeRequest))
+            {
+                break;
+            }
+            if (writeRequest->cancel())
+            {
+                std::cout << "Cancelled." << std::endl;
+                break;
+            }
+            if (writers->size() > 0)
+            {
+                int x = 0;                
+                if (writers->size() != writeRequest->values().values_size())
+                {
+                    std::cout << "ERROR: The number of writers and values read did not match" << std::endl;
```

Comment:
Should this return error to client instead? Is that even possible?

Reply:
Will be working on if it's feasible in a follow-up with other changes.